src/nevermined/Assets.ts

```diff
@@ -115,16 +146,47 @@ export class Assets extends Instantiable {
     }
 
     /**
-     * Returns a DDO by DID.
+     * Returns a DDO by DID. Depending of the resolution policy it prioritize the Metadata API or Immutable urls.
      * @param did - Decentralized ID.
+     * @param policy - It specifies the resolve policy to apply. It allows to select that priorities during the asset resolution via Metadata API or Immutable URLs (IPFS, Filecoin, etc)
      * @returns {@link DDO}
      */
-    public async resolve(did: string): Promise<DDO> {
-        const { serviceEndpoint } =
+    public async resolve(did: string, policy: DIDResolvePolicy = DIDResolvePolicy.ImmutableFirst): Promise<DDO> {
+        const { serviceEndpoint, immutableUrl } =
             await this.nevermined.keeper.didRegistry.getAttributesByDid(did)
-        return this.nevermined.metadata.retrieveDDOByUrl(serviceEndpoint)
+
+        const immutableUrlValid = immutableUrl && immutableUrl.length > 10
```

Comment:
this should be handled in `retrieveDDOFromExternalBackend`

Reply:
refactored